orangecontrib/text/widgets/utils/widgets.py

```diff
@@ -0,0 +1,111 @@
+import os
+
+from PyQt4.QtGui import (QComboBox, QWidget, QHBoxLayout, QPushButton, QStyle,
+                         QSizePolicy, QFileDialog)
+from PyQt4 import QtCore
+
+
+class ComboBox(QComboBox):
+    def __init__(self, master, attr, items):
+        super().__init__()
+        self.attr = attr
+        self.master = master
+
+        if not isinstance(items[0], tuple):
+            self.items = [(str(item).capitalize(), item) for item in items]
+        else:
+            self.items = items
+
+        for i, (key, value) in enumerate(self.items):
+            self.addItem(key)
+            if value == getattr(master, attr, None):
+                self.setCurrentIndex(i)
+
+        self.currentIndexChanged[int].connect(self.syncronize)
+
+    def syncronize(self, i):
+        setattr(self.master, self.attr, self.items[i][1])
+
+
+class FileWidget(QWidget):
```

Comment:
It's very nice that you put this into one common place 👍  Could you also refactor `owloadcorpus.py` to use this instead? I think most of its functionality is the same as here.